server/src/lib/basket.ts

```diff
@@ -11,80 +11,154 @@ export const API_URL =
     ? 'https://basket.mozilla.org'
     : 'https://basket-dev.allizom.org';
 
+const basketConfig = {
+  uri: API_URL + '/news/common-voice-goals/',
+  method: 'POST',
+  headers: {
+    'x-api-key': BASKET_API_KEY,
+  },
+};
+
 function toISO(date: string) {
   return date ? new Date(date).toISOString().slice(0, -5) + 'Z' : null;
 }
 
-export async function sync(client_id: string) {
-  await computeGoals(client_id);
-
+async function getCurrentStatus(client_id: string) {
   const [[row]] = await db.query(
+    `SELECT u.*,
+      NOW() AS new_last_active,
+      HOUR(TIMEDIFF(NOW(), u.last_active)) AS hours_elapsed
+    FROM user_client_newsletter_prefs AS u
+    WHERE client_id = ?`,
+    [client_id]
+  );
+
+  return row;
+}
+
+function updateLastActive(currentUserStats: any) {
```

Comment:
This is a pretty complex object; would be nice to have a type definition for it.